.github/workflows/goreleaser.yml

```
@@ -0,0 +1,46 @@
+on:
+  workflow_call:
+    inputs:
+      go-version:
+        required: false
+        type: string
+        default: 1.20
+    secrets:
+      GORELEASER_PAT:
+        required: true
+
+  goreleaser:
+    name: Upload Assets To Github w/ goreleaser
+    runs-on: ubuntu-latest
+    needs: create_release
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ inputs.go-version }}
+          check-latest: true
+      - name: Install cosign
+        uses: sigstore/cosign-installer@v2
+        with:
+          cosign-release: 'v1.13.1'
+      - name: Get Release Date
+        id: release_date
+        run: |
+          RELEASE_DATE=$(date +"%y-%m-%d")
```

We should enforce UTC, although the server is probably on it:
```
```suggestion
          RELEASE_DATE=$(date -u +"%y-%m-%d")
```